js-code-guidelines.md

```diff
@@ -73,39 +68,79 @@ For the majority of our JavaScript projects, our goals are to:
+## Tech Leads and Lead Maintainers
 
-Please follow the conventions described in this document.
+> This section is a formalization of the proposal and discussion around the [Module Lead Maintainers - Sharing the Responsibility over the IPFS module base](https://github.com/ipfs/pm/issues/600) proposal.
 
-When reporting a bug, if possible, provide a way for us to reproduce it (or even better, write a test that fails with your case).
+We have two types of leads in the JavaScript project ecosystems, the Tech Lead and the Module Lead Maintainer. A brief description of both roles is:
 
-Always run tests before pushing and PR'ing your code.
+- A Tech Lead directs the development of an entire ecosystem of modules (i.e js-ipfs, js-libp2p, js-ipld and js-multiformats), it has a complete understanding of the stack, the IPFS project, its goals and participates actively in the ROADMAP planning.
+- A Lead Maintainer is a contributor that has shown extraordinary ability to contribute to the project and willingness to make the project better by taking the responsibility of stewarding one of its modules forward.
+
+With this structure, we expect to achieve the following goals:
+
+- Recognize extraordinary contributions and empower contributors to take an even more important role in the project.
+- Reduce PR review time and Time To Release.
+- Increase the overall quality of the project.
+- Help users know who to reach out to for help.
+- Have a clear protocol to pass on the baton.
 
-### Captains and Maintainers
+The current Tech Leads are:
 
-For each JavaScript repository in the IPFS, libp2p, IPLD, or related GitHub orgs, there should be a single Captain specified in the Maintainers section of the README for that repository. The Captain is in charge of merging PRs, keeping issues up to date, and overall quality of a repository.
+- [David Dias](https://github.com/diasdavid/) for js-ipfs, js-libp2p js-multiformats ecosystems.
+- [Volker Mische](https://github.com/vmx) for the js-ipld ecosystem.
 
-Sometimes, a Captain may elect to have other maintainers that also have merge ability and commit access to the main repo. These maintainers can help out, but defer to the Captain as the person in charge of maintaining quality across the repo. It is important that this distinction is explicit; if there are long-standing PRs or issues, it is ultimately up to the Captain to gather information about the issue or PR. A Captain only makes a decision if he _needs_ to and _all_ methods of discussion are exhausted. Our community strives to trust the Captain as someone who ultimately has the most knowledge of a repo (even if they are also opinionated, and even if they have to spend effort to source that knowledge). This may change in the future, if we go with more non-hierarchical model.
+The current Lead Maintainers can be identified either by the `maintainer` field in the package.json of the module and/or the section `Maintainer` in the README.md of the module.
```

Comment:
Also watch for typos as `maintainers` field exists but is overwritten/handled by npm.

Ref: The "maintainers" field in the top-level of the registry metadata (ie, not versioned) is the npm-controlled list of the npm usernames of the people with permission to write to that package. This can be modified via the npm owners command. (Confusing, I know.)
https://github.com/npm/www/issues/133#issuecomment-284906561

Reply:
thanks for catching this. Changed to `lead-maintainer` instead :)